pkg/apiserver/iam/im.go

```diff
@@ -35,7 +35,8 @@ import (
 )
 
 const (
-	emailRegex = "^[a-z0-9]+([._\\-]*[a-z0-9])*@([a-z0-9]+[-a-z0-9]*[a-z0-9]+.){1,63}[a-z0-9]+$"
+	// RFC 5322 regex, https://tools.ietf.org/html/rfc5322
+	emailRegex = "^[a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+(?:\\.[a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-zA-Z0-9](?:[a-zA-Z0-9-]*[a-zA-Z0-9])?\\.)+([a-zA-Z]{2,})$"
```

Comment:
why we change to this ? Also I find this expression can't really match email address. I tested it on https://www.regextester.com/, the older expression do match some though. Please double check.

Reply:
An email address may contain upper-case letters as defined in RFC5322 (http://tools.ietf.org/html/rfc5322#section-3.4.1).